src/v0/destinations/tune/transform.js

```diff
@@ -0,0 +1,97 @@
+const get = require('get-value');
+const { InstrumentationError } = require('@rudderstack/integrations-lib');
+const { defaultRequestConfig, simpleProcessRouterDest, getHashFromArray } = require('../../util');
+
+const responseBuilder = (message, { Config }) => {
+  const { tuneEvents } = Config; // Extract tuneEvents from config
+  const { properties, event: messageEvent } = message; // Destructure properties and event from message
+
+  // Find the relevant tune event based on the message's event name
+  const tuneEvent = tuneEvents.find((event) => event.eventName === messageEvent);
+
+  if (tuneEvent) {
+    const standardHashMap = getHashFromArray(tuneEvent.standardMapping);
+    const advSubIdHashMap = getHashFromArray(tuneEvent.advSubIdMapping);
+    const advUniqueIdHashMap = getHashFromArray(tuneEvent.advUniqueIdMapping);
+
+    const mapPropertiesWithNestedSupport = (msg, mappings) => {
+      const newParams = {}; // Create a new object for parameters
+      Object.entries(mappings).forEach(([key, value]) => {
+        let data; // Declare data variable
+
+        if (key.split('.').length > 1) {
+          // Handle nested keys
+          data = get(msg, key); // Use `get` to retrieve nested data
+          if (data) {
+            newParams[value] = data; // Map to the corresponding destination key
+          }
+        } else {
+          // Handle non-nested keys
+          data = get(properties, key); // Retrieve data from properties directly
+          if (data) {
+            newParams[value] = data; // Map to the corresponding destination key
+          }
+        }
+      });
+      return newParams; // Return the new params object
+    };
```

Comment:
Simplified logic
```suggestion
    const mapPropertiesWithNestedSupport = (msg, mappings) => {
      const mappedObj = {}; // Create a new object for parameters
      Object.entries(mappings).forEach(([key, value]) => {
        let data; // Declare data variable
        let args = { object: properties, key };
        if (key.split('.').length > 1) {
          // Handle nested keys
          args.object = msg;
        }
        data = get(msg, key); // Use `get` to retrieve nested data
        if (data) {
          mappedObj[value] = data; // Map to the corresponding destination key
        }
      });
      return mappedObj; // Return the new params object
    };
```

Reply:
Are we also sure `key` can never be any other type than `string` ?

Reply:
yes, it will be in string only. Also for betterment I've converted key to string.